src/Farmer/Builders/Builders.Databricks.fs

```diff
@@ -0,0 +1,143 @@
+    /// Sets the encryption key configuration
+    [<CustomOperation "encryption_key">]
+    member _.EncryptionKey (state:WorkspaceConfig, keyName, ?keyVersion) =
+        let keyVersion = defaultArg keyVersion "latest"
+        let encryption =
+            state.Encryption
+            |> Option.map(fun encryptionConfig ->
+                { encryptionConfig with
+                    KeyName = keyName
+                    KeyVersion = keyVersion})
+            |> Option.orElse
+                (Some { KeyVault = ResourceName.Empty
+                        KeyName = keyName
+                        KeyVersion = keyVersion
+                        KeySource = "Microsoft.Keyvault" })
+        { state with
+            PrepareEncryption = Enabled
+            Encryption = encryption }
+
+    /// Sets the vnet
+    [<CustomOperation "byov_vnet">]
+    member _.ByovVnet (state:WorkspaceConfig, vnet:ResourceName) =
+        let config =
+            state.ByovConfig
+            |> Option.map(fun vnetConfig -> { vnetConfig with Vnet = External(Unmanaged(virtualNetworks.resourceId vnet)) })
+            |> Option.orElse
+                (Some { Vnet = External(Unmanaged(virtualNetworks.resourceId vnet))
+                        PublicSubnet = ResourceName.Empty
+                        PrivateSubnet = ResourceName.Empty })
+        { state with ByovConfig = config }
+    member this.ByovVnet(state:WorkspaceConfig, name:string) = this.ByovVnet(state, ResourceName name)
+    member this.ByovVnet(state:WorkspaceConfig, vnet:Arm.Network.VirtualNetwork) = this.ByovVnet(state, vnet.Name)
+    member this.ByovVnet(state:WorkspaceConfig, vnet:VirtualNetworkConfig) = this.ByovVnet(state, vnet.Name)
+
+    /// Set the existing public subnet
+    [<CustomOperation "byov_public_subnet">]
+    member _.ByovPublicSubnet (state:WorkspaceConfig, publicSubnet:ResourceName) =
+        let config =
+            state.ByovConfig
+            |> Option.map(fun vnetConfig -> { vnetConfig with PublicSubnet = publicSubnet })
+            |> Option.orElse
+                (Some { Vnet = External(Unmanaged(virtualNetworks.resourceId ResourceName.Empty))
+                        PublicSubnet = publicSubnet
+                        PrivateSubnet = ResourceName.Empty })
+        { state with ByovConfig = config}
+    member this.ByovPublicSubnet(state:WorkspaceConfig, publicSubnet) = this.ByovPublicSubnet(state, ResourceName publicSubnet)
+
+    /// Sets the existing private subnet
+    [<CustomOperation "byov_private_subnet">]
+    member _.ByovPrivateSubnet (state:WorkspaceConfig, privateSubnet:ResourceName) =
+        let config =
+            state.ByovConfig
+            |> Option.map(fun vnetConfig -> { vnetConfig with PrivateSubnet = privateSubnet })
+            |> Option.orElse
+                (Some { Vnet = External(Unmanaged(virtualNetworks.resourceId ResourceName.Empty))
```

Comment:
I think this code is problematic. External Unmanaged means a resource that is created *outside* of Farmer. However, if you are creating a VNet as part of the Farmer template, it'll need to be specified as External Managed, and you'll need to provide this as a dependency onto the workspace to ensure that the VNet is created *before* the workspace.

Have a look at how some other resources do this - if you get stuck or it's not going, no worries - let me know, one of me or the team will get this over the line.

Reply:
I tried looking at how a Virtual Machine is linked to a Virtual Network but I couldn't work out how to handle all cases such as the VNet already existing or the VNet being deployed by Farmer. I wasn't totally happy with this part and knew that it didn't handle all of the cases

Reply:
Let me have a bash at this - I think there are a few combinations that need to be considered.